cli/src/commands/router/commands/compose.ts

```diff
@@ -94,14 +94,21 @@ export default (opts: BaseCommandOptions) => {
     const routerConfig = buildRouterConfig({
       argumentConfigurations: result.federationResult.argumentConfigurations,
       federatedSDL: printSchema(result.federationResult.federatedGraphSchema),
-      subgraphs: config.subgraphs.map((s, index) => ({
-        id: `${index}`,
-        name: s.name,
-        url: normalizeURL(s.routing_url),
-        sdl: sdls[index],
-        subscriptionUrl: s.subscription?.url ?? s.routing_url,
-        subscriptionProtocol: s.subscription?.protocol ?? 'ws',
-      })),
+      subgraphs: config.subgraphs.map((s, index) => {
+        const subgraphConfig = result.federationResult!.subgraphConfigBySubgraphName.get(s.name);
+        const schema = subgraphConfig?.schema;
+        const configurationDataMap = subgraphConfig?.configurationDataMap;
```

Comment:
Could you describe briefly what this data structure is used for?

Reply:
`schema` is the normalized GraphQLSchema (graphql.js) that originally comes from `normalizeSubgraph`.
`ConfigurationDataMap` is the intermediate representation of the engine config that is turned into proto.

Before, both of these structures were returned by calling `normalizeSubgraph` again in `buildRouterConfig`, but now we normalize once and pass the schema and config up to the federator, and finally to `buildRouterConfig`.

Reply:
`subgraphConfig` is just a container for the schema and the ConfigurationDataMap. It's mapped by subgraph name so we can keep track of which schema and config is for which graph (because this returned from batch normalize).

Reply:
Could you add these as comments to the types in composition lib?